### PR TITLE
Qt6 porting

### PR DESCRIPTION
**Pull request by cryos**

### CODE REVIEW

File: avogadro/qtgui/filebrowsewidget.cpp [hunk 1]
```diff
@@ -159,15 +159,15 @@
   if (!env.contains(QStringLiteral("PATH")))
     return result;
 
-  static QRegExp pathSplitter = QRegExp(
+  static QString pathSplitter =
```
[the remark this answers is not in the document]
Comment (cryos) [reply avogadro/qtgui/filebrowsewidget.cpp:162]: Just porting code, so I am tempted to leave this for now.

File: avogadro/rendering/solidpipeline.cpp [hunk 2]
```diff
@@ -41,35 +84,47 @@
                          *texDepth, 0);
 }
 
+SolidPipeline::SolidPipeline()
+  : m_pixelRatio(1.0f), m_aoEnabled(true), m_aoStrength(1.0f),
+    m_edEnabled(true), m_edStrength(1.0f), m_width(0), m_height(0),
+    d(new Private)
```
[the remark this answers is not in the document]
Comment (cryos) [reply avogadro/rendering/solidpipeline.cpp:90]: Rusty and missed this the first time, someone else's code, but added a simple copy constructor to account for the d pointer addition. Hopefully done with changes before landing the initial port to Qt 6.

[the remark this answers is not in the document]
Comment (cryos) [reply avogadro/rendering/solidpipeline.cpp:90]: This is a duplicate of the one below I fixed, bug in the action code?

**Comment by cryos** [comment]
OK, I will let this build and if nothing new is spotted I intend to merge later today. I would like to then create smaller PRs to focus on other things as this is already getting a little bigger than I wanted.

**Comment by ghutchis** [comment]
The error on Mac is:
```
CMake Error at /opt/homebrew/Cellar/cmake/3.25.2/share/cmake/Modules/BundleUtilities.cmake:244 (message):
  BundleUtilities cannot be included at configure time!
Call Stack (most recent call first):
  cmake/DeployQt5.cmake:114 (include)
  avogadro/lastinstall/CMakeLists.txt:64 (include)
```

That may be a chicken-and-egg issue with the avogadroapp patch.

**Comment by ghutchis** [comment]
The static analysis fail is also CMake:
```
CMake Error at avogadro/qtgui/CMakeLists.txt:109 (qt_wrap_ui):
  qt_wrap_ui called with incorrect number of arguments


CMake Error at avogadro/qtgui/CMakeLists.txt:115 (qt_add_resources):
  Unknown CMake command "qt_add_resources".
```

**Comment by ghutchis** [comment]
I'm not sure that either of those are critical at the moment, so I'm okay to merge.

**Comment by cryos** [comment]
Yeah, the deploy thing looks odd, the static analysis may be the Qt is too old. The unversioned CMake commands were added in Qt 5.15, I will merge and we can pick up the pieces. This was always going to require a little coordination and maybe some fixing as it is a pretty sizeable update.

**Pull request merged**